src/sherpa_ai/database/user_usage_tracker.py

```diff
@@ -69,17 +77,44 @@ def get_whitelist_by_user_id(self, user_id):
 
     def is_in_whitelist(self, user_id):
         return bool(self.get_whitelist_by_user_id(user_id))
+    
+    def add_and_check_data(self, combined_id, token, reset_timestamp=False , reminded_timestamp=False):
+        self.add_data(combined_id=combined_id , token=token, reset_timestamp=reset_timestamp, reminded_timestamp=reminded_timestamp)
+        self.token_left_calculator(combined_id=combined_id)
 
-    def add_data(self, combined_id, token, reset_timestamp=False):
+    def add_data(self, combined_id, token, reset_timestamp=False , reminded_timestamp=False):
         data = UsageTracker(
             user_id=combined_id,
             token=token,
             timestamp=int(time.time()),
             reset_timestamp=reset_timestamp,
+            reminded_timestamp=reminded_timestamp
         )
         self.session.add(data)
         self.session.commit()
 
+    def percentage_used(self , combined_id):
+        total_token_since_last_reset = self.get_sum_of_tokens_since_last_reset(
+                user_id=combined_id
+            )
+        return  (total_token_since_last_reset*100)/self.max_daily_token 
+
+
+    def token_left_calculator(self, combined_id):
```

Comment:
Rename this function to say what it does. For example, remind_user_of_daily_token_limit